mlflow/R/mlflow/R/model-h2o.R

```diff
@@ -0,0 +1,88 @@
+#' @rdname mlflow_save_model
+#' @export
+mlflow_save_model.H2OModel <- function(model,
+                                       path,
+                                       model_spec = list(),
+                                       conda_env = NULL,
+                                       ...) {
+  h2o_assert_installed()
+
+  path <- normalizePath(path)
+  if (dir.exists(path)) unlink(path, recursive = TRUE)
+  dir.create(path, recursive = TRUE)
+
+  model_data_subpath <- "model.h2o"
+  model_data_path <- file.path(path, model_data_subpath)
+
+  dir.create(model_data_path)
+
+  h2o_save_location <- h2o::h2o.saveModel(
+    object = model, path = model_data_path, force = TRUE
+  )
+  model_file <- basename(h2o_save_location)
+
+  settings <- list(
+    full_file = h2o_save_location,
+    model_file = model_file,
+    model_dir = model_data_path
+  )
+  yaml::write_yaml(settings, file.path(model_data_path, "h2o.yaml"))
+
+  version <- as.character(utils::packageVersion("h2o"))
+
+  conda_env <- if (!is.null(conda_env)) {
```

Comment:
This also exists in xgboost. I think we could factor this out into a function and use it in both xgboost and h2o and potentially others later.